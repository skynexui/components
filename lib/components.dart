library skynexui_components;

export 'package:flutter/material.dart'
    hide Typography, Text, TextField, Image, Theme;
export 'package:provider/provider.dart';
export 'components/provider/provider.dart';
export 'components/box/box.dart';
export 'components/text/text.dart';
export 'components/flatlist/flatlist.dart';
export 'core/breakpoints/breakpoints.dart';
export 'core/stylesheet/stylesheet.dart';
