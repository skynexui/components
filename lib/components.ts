export { Box } from '@lib/components/box/box';
export { Provider, useTheme } from '@lib/components/provider/provider';
export { Text } from '@lib/components/text/text';
export { GridDisplay } from '@lib/components/griddisplay/griddisplay';
export { TextField } from '@lib/components/textfield/textfield';
export { Icon } from '@lib/components/icon/icon';
export { Button } from '@lib/components/button/button';
export { Image } from '@lib/components/image/image';

// Tooling
export { ThemeGenerator } from '@lib/core/theme/generator/generator';
