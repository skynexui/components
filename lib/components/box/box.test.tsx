import React from 'react';
import { Box } from '@lib/components';
import { render } from '@lib/utils/test/rtl-utils';

it('works', () => {
  const { container } = render(
    <Box
      styleSheet={{
        backgroundColor: {
          'Breakpoints.xs': 'red',
        },
        padding: {
          'Breakpoints.xs': 10,
        },
      }}
    >
      Hello a
    </Box>
  );
  expect(container).toMatchSnapshot();
});
