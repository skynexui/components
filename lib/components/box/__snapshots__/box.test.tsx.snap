// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`works 1`] = `
<div>
  <main>
    <div
      class="sknui-box  "
    >
      Hello a
    </div>
    <style
      jsx="true"
    >
      
          div.sknui-box {
            background-color:red;padding:10;;
          }
          div.sknui-box:hover {
            ;
          }
          div.sknui-box:focus {
            ;
          }
          @media screen and (min-width: 480px) {
            div.sknui-box {
              ;
            }
            div.sknui-box:hover {
              ;
            }
            div.sknui-box:focus {
              ;
            }
          }
          @media screen and (min-width: 768px) {
            div.sknui-box {
              ;
            }
            div.sknui-box:hover {
              ;
            }
            div.sknui-box:focus {
              ;
            }
          }
          @media screen and (min-width: 992px) {
            div.sknui-box {
              ;
            }
            div.sknui-box:hover {
              ;
            }
            div.sknui-box:focus {
              ;
            }
          }
          @media screen and (min-width: 1200px) {
            div.sknui-box {
              ;
            }
            div.sknui-box:hover {
              ;
            }
            div.sknui-box:focus {
              ;
            }
          }
        
    </style>
  </main>
</div>
`;
